src/otx/algo/utils/xai_utils.py

```diff
@@ -18,19 +20,22 @@
 def get_processed_saliency_maps(
     explain_hook: BaseRecordingForwardHook,
     explain_config: ExplainConfig,
-    predictions: list | None,
+    predictions: list[Any] | list[OTXBatchPredEntityWithXAI | InstanceSegBatchPredEntityWithXAI] | None,
     work_dir: Path | None,
 ) -> list:
     """Implement saliency map filtering and post-processing."""
     # Optimize for memory <- TODO(negvet)
-    raw_saliency_maps = explain_hook.records
+    raw_saliency_maps: list = []
+
+    if predictions is not None:
+        raw_saliency_maps = predictions[0].saliency_maps
```

Comment:
Do I understand correctly that we can have only 1 element in `OTXBatchPredEntityWithXAI`, that's why it's safe to do `predictions[0]` omitting all other predictions?

Reply:
Predictions store per-batch explain result as a list of maps.
I assume this function to be deleted asap in the upcoming PRs, and replaced with a configurable XAI artifacts manager.